web/packages/teleport/src/Assist/context/AssistContext.tsx

```diff
@@ -465,30 +467,27 @@ export function AssistContextProvider(props: PropsWithChildren<unknown>) {
 
           break;
 
-        case MessageTypeEnum.SESSION_END:
-          // we don't know the nodeId of the session that ended, so we have to
-          // count the finished sessions and then mark them all as done once
-          // they've all finished
+        case MessageTypeEnum.SESSION_END: {
+          const data = JSON.parse(msg.payload) as SessionData;
+
+          dispatch({
+            type: AssistStateActionType.FinishCommandResult,
+            conversationId: state.conversations.selectedId,
+            commandResultId: nodeIdToResultId.get(data.session.server_id),
+          });
+
           sessionsEnded += 1;
 
           if (sessionsEnded === nodeIdToResultId.size) {
             const message = proto.encodeCloseMessage();
             const bytearray = new Uint8Array(message);
 
-            for (const nodeId of nodeIdToResultId.keys()) {
-              dispatch({
-                type: AssistStateActionType.FinishCommandResult,
-                conversationId: state.conversations.selectedId,
-                commandResultId: nodeIdToResultId.get(nodeId),
-              });
-
-              executeCommandWebSocket.current.send(bytearray.buffer);
-            }
-
+            executeCommandWebSocket.current.send(bytearray.buffer);
```

Comment:
I think this line is no more needed. I forgot to remove it while I was working on this https://github.com/gravitational/teleport/pull/28456